src/stream/mod.rs

```diff
@@ -280,30 +238,21 @@ impl Stream {
                 tokio::fs::remove_file(path.as_ref()).await?;
                 Err(e)
             }
-        };
+        }.map(|_| {
+            let mut pathbuf = PathBuf::new();
+            pathbuf.push(path);
+            pathbuf
```

Comment:
`Path` has a [`to_path_buf`](https://doc.rust-lang.org/std/path/struct.Path.html#method.into_path_buf) method.

Reply:
`error[E0599]: no method named 'to_path_buf' found for type parameter 'P' in the current scope`. Same goes for `into_path_buf`.

Reply:
That's because `P` is a generic type, and everything we know about `P` is, that it implements `AsRef<Path>`. So an instance of `P` is not a `Path`, but it allows us to get a reference to an instance of a `Path` by calling `.as_ref()`.
In this case, you can call `path.as_ref().to_path_buf()`.